Symfony/CS/Fixer/PSR2/EofEndingFixer.php

```diff
@@ -25,13 +26,25 @@ public function fix(\SplFileInfo $file, $content)
     {
         // [Structure] A file must always end with a linefeed character
 
-        $content = rtrim($content);
+        if (empty($content)) {
+            return $content;
+        }
+
+        $tokens = Tokens::fromCode($content);
+        $count = $tokens->count();
+        if (0 === $count) {
+            return;
```

Comment:
To be honest I don't think it is possible. I've added the `empty` check to satisfy the existing unit test for empty files. I can update the code to remove the zero tokens check.

However IMO this whole test and support for not adding a line break in an empty file should go. I don't see why someone would add an empty PHP file to their code base and even so would run it through a CS fixer and expect it not to be changed up to the PSR standard (that states the line break should be there).
This would be a BC break so Im happy to hear what others think.